### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by El-Potato-Slayer**
- Created the `logic` folder
- Installed Big.js
- Created the `calculate.js` and `operate.js` files

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -2,8 +2,11 @@ import { Fragment } from 'react';
 import '../App.css';
 import ButtonPanel from './ButtonPanel';
 import Display from './Display';
+import calculate from '../logic/calculate';
 
 function App() {
+  calculate();
+
```
Comment (Elbie-em) [inline src/components/App.js:9]: Only import the calculate module. Calling the function with no arguments as you have done is not required in this milestone.

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+export default function calculate(calcData, buttonName) {
```
Comment (Elbie-em) [inline src/logic/calculate.js]: There is no import of operate.js, kindly add this.

File: src/logic/calculate.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  // I have no idea what the 'next' properties is supposed to be. Mind providing an explanation?
+  const data = calcData;
+
```
Comment (Elbie-em) [inline src/logic/calculate.js:6]: - [ ] The calcData should have three properties {total, next, operation}. 'next' is basically the next value typed in the calculator or the previous value from the total when attempting multiple operations.

File: src/logic/calculate.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    data.next *= -1;
+  }
+  if (buttonName === '%') {
+    data.total = (data.total * 0.01).toString();
+  }
+  if (buttonName === '=') {
+    // Need to understand what next is before I can implement all conditionals
+  }
+  return data;
+}
```
Comment (Elbie-em) [inline src/logic/calculate.js:22]: - [ ] Add the operation property to your cases in order for your to calculate function to produce proper results.

File: src/logic/calculate.js (hunk 3 above)
Comment (Elbie-em) [reply src/logic/calculate.js:6]: - [ ] You can destructure your calcData to make your code more readable.

File: src/logic/operate.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import Big from 'big.js';
+
+export default function operate(numberOne, numberTwo, operation) {
+  let output;
+  if (operation === '+') {
```
Comment (Elbie-em) [inline src/logic/operate.js:7]: - [ ] both numberOne and numberTwo should be wrapped in big.js, here is how you could handle it.
const num1 = Big(numberOne)
const num2 = Big(numberTwo)

File: src/logic/operate.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    output = Big(numberOne).minus(Big(numberTwo));
+  }
+  if (operation === '÷') {
+    output = Big(numberOne).div(Big(numberTwo));
+  }
+  if (operation === '×') {
+    output = Big(numberOne).mul(Big(numberTwo));
+  }
+  return output;
+}
```
Comment (Elbie-em) [inline src/logic/operate.js:23]: - [ ] operate should switch operation and use plus, minus, times, div and modifier(
turning the number into a percentage based number, `e.g. 81 % == .81`)

**Review by Elbie-em — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED 🟡  ♻️ 

Hi @El-Potato-Slayer, you have really made a great effort in completing the other milestones and getting this far. However before this milestone is approved, there are a couple of changes you need to make in regards to the requirements. Check my comments below ⬇️ 

*Please apply the changes requested and request another code review. All the best 💪 *

> Elbie Moonga (Part-time TSE)

> Github (@Elbie-em)
> Slack (Elbie Moonga)
> Twitter ([@ElbieEm](https://twitter.com/ElbieEm))
> LinkedIn([Elbie Moonga](https://www.linkedin.com/in/elbiemoonga/))

**Comment by rukundoeric** [comment]
# Status: 🍾 PROJECT APPROVED 🍾
_Excellent job on this project guys. 🎆 Keep it up._

- ### To Highlight
  - Follows gitwork-flow.  💯
  - Codebase looks clean and professional. 💯 
  

- ### To Improve
  - N/A

_Congratulations on the splendid job **@team** 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/),
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/),
Slack: Rukundo Eric**

**Pull request merged**